apps/events-helsinki/src/domain/app/getEventsStaticProps.ts

```diff
@@ -96,36 +84,82 @@ type GetGlobalCMSDataParams = {
   context: GetStaticPropsContext;
 };
 
+type ReturnedGlobalCMSData = {
+  headerMenu?: Menu;
+  footerMenu?: Menu;
+  languages?: Language[];
+};
+
 // Get CMS data that's required on every page
-async function getGlobalCMSData({ client, context }: GetGlobalCMSDataParams) {
-  const menuLocation = getMenuLocationFromLanguage(
-    getLocaleOrError(context.locale)
-  );
-  const queryOptions = {
-    query: GLOBAL_QUERY,
+async function getGlobalCMSData({
+  client,
+  context,
+}: GetGlobalCMSDataParams): Promise<ReturnedGlobalCMSData> {
+  const locale: AppLanguage = (context?.locale ?? 'fi') as AppLanguage;
+  const headerNavigationMenuName = DEFAULT_HEADER_MENU_NAME[locale];
+  const { data: headerMenuData } = await client.query({
+    query: MenuDocument,
     variables: {
-      menuLocation,
+      id: headerNavigationMenuName,
+      // idType: 'URI'
     },
-  };
-  const { data } = await client.query(queryOptions);
+  });
   client.writeQuery({
-    ...queryOptions,
+    query: MenuDocument,
+    variables: {
+      id: headerNavigationMenuName,
+    },
     data: {
-      ...data,
-      // Only use languages that this project supports
-      pageLanguages: getSupportedLanguages(data.pageLanguages, context),
+      ...headerMenuData,
+    },
+  });
+
+  const footerNavigationMenuName = DEFAULT_FOOTER_MENU_NAME[locale];
+  const { data: footerMenuData } = await client.query({
+    query: MenuDocument,
+    variables: {
+      id: footerNavigationMenuName,
+      // idType: 'URI'
     },
   });
+  client.writeQuery({
+    query: MenuDocument,
+    variables: {
+      id: footerNavigationMenuName,
+    },
+    data: {
+      ...footerMenuData,
+    },
+  });
+
+  const { data: languagesData } = await client.query({
+    query: LanguagesDocument,
+  });
+
+  client.writeQuery({
+    query: LanguagesDocument,
+    data: { ...languagesData },
+  });
```

Comment:
These adds the content to the Apollo cache. However, I don't think we get a big advantage of this, if any, because the content is now served as component props instead of hook usage.